hist/hist/src/TH1.cxx

```diff
@@ -9928,6 +9928,23 @@ TH1F::~TH1F()
 {
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Increment bin content by 1.
+
+void TH1F::AddBinContent(Int_t bin)
+{
+   if (fArray[bin] < 1e7) ++fArray[bin];
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Increment bin content by w.
+
+void TH1F::AddBinContent(Int_t bin, Double_t w)
+{
+   const Float_t newval = fArray[bin] + Float_t(w);
```

Comment:
This is not correct. A float can be much larger that 1.E7. In case of a weighted histograms one should check if w/fArray[bin] is smaller than std::numeric_limits<float>::epsilon()

Reply:
Shouldn't we instead check?

if (std::nextafter(fArray[bin]) > fArray[bin] + Float_t(w)); then set rounding-error flag internally ?
Epsilon does not account for this dynamically.

Reply:
`if (std::abs(std::nexttowardf(fArray[bin], w + fArray[bin]) - fArray[bin]) <= std::abs(w)) fArray[bin] += Float_t(w);` should work, but it's a lot of overhead in my opinion

Reply:
But this is too involved in my opinion and delays things... i think we should just do the check for integer as it's comparing with a fixed value.